@snicoll Thanks for your work!!
I try this change in my local. The AOT mode on JVM work fine, but native mode does not work...

p.s.
Following is result that I've applied the Spring Boot 2.6.3 and spring-native 0.11.2-SNAPSHOT latest version.

```
...
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'runner': Unsatisfied dependency expressed through method 'runner' parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'cityMapper': Error setting property values; nested exception is org.springframework.beans.NotWritablePropertyException: Invalid property 'mapperInterface' of bean class [org.mybatis.spring.mapper.MapperFactoryBean]: Bean property 'mapperInterface' is not writable or has an invalid setter method. Does the parameter type of the setter match the return type of the getter?
        at org.springframework.aot.beans.factory.InjectedConstructionResolver.resolve(InjectedConstructionResolver.java:88) ~[na:na]
        at org.springframework.aot.beans.factory.InjectedElementResolver.resolve(InjectedElementResolver.java:19) ~[na:na]
        at org.springframework.aot.beans.factory.InjectedElementResolver.create(InjectedElementResolver.java:50) ~[na:na]
        at org.springframework.aot.beans.factory.BeanDefinitionRegistrar$InstanceSupplierContext.create(BeanDefinitionRegistrar.java:193) ~[na:na]
        at org.mybatis.spring.nativex.sample.simple.ContextBootstrapInitializer.lambda$registerMybatisSpringNativeSampleApplication_runner$1(ContextBootstrapInitializer.java:11) ~[na:na]
        at org.springframework.aot.beans.factory.ThrowableFunction.apply(ThrowableFunction.java:18) ~[na:na]
        at org.springframework.aot.beans.factory.BeanDefinitionRegistrar.lambda$instanceSupplier$0(BeanDefinitionRegistrar.java:97) ~[na:na]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.obtainFromSupplier(AbstractAutowireCapableBeanFactory.java:1249) ~[na:na]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1191) ~[na:na]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:582) ~[na:na]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542) ~[na:na]
        at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335) ~[na:na]
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[na:na]
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333) ~[na:na]
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208) ~[na:na]
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:953) ~[na:na]
        at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:918) ~[na:na]
        at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:583) ~[na:na]
        at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:732) ~[mybatis-spring-native-sample-simple:2.6.3]
        at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:414) ~[mybatis-spring-native-sample-simple:2.6.3]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:302) ~[mybatis-spring-native-sample-simple:2.6.3]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1303) ~[mybatis-spring-native-sample-simple:2.6.3]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1292) ~[mybatis-spring-native-sample-simple:2.6.3]
        at org.mybatis.spring.nativex.sample.simple.MybatisSpringNativeSampleApplication.main(MybatisSpringNativeSampleApplication.java:32) ~[mybatis-spring-native-sample-simple:0.0.1-SNAPSHOT]
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'cityMapper': Error setting property values; nested exception is org.springframework.beans.NotWritablePropertyException: Invalid property 'mapperInterface' of bean class [org.mybatis.spring.mapper.MapperFactoryBean]: Bean property 'mapperInterface' is not writable or has an invalid setter method. Does the parameter type of the setter match the return type of the getter?
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyPropertyValues(AbstractAutowireCapableBeanFactory.java:1744) ~[na:na]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1452) ~[na:na]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:619) ~[na:na]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:542) ~[na:na]
        at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335) ~[na:na]
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[na:na]
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333) ~[na:na]
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208) ~[na:na]
        at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:276) ~[na:na]
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1389) ~[na:na]
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1309) ~[na:na]
        at org.springframework.aot.beans.factory.InjectedConstructionResolver.lambda$resolve$0(InjectedConstructionResolver.java:83) ~[na:na]
        at org.springframework.aot.beans.factory.InjectedConstructionResolver.resolveDependency(InjectedConstructionResolver.java:97) ~[na:na]
        at org.springframework.aot.beans.factory.InjectedConstructionResolver.resolve(InjectedConstructionResolver.java:83) ~[na:na]
        ... 23 common frames omitted
Caused by: org.springframework.beans.NotWritablePropertyException: Invalid property 'mapperInterface' of bean class [org.mybatis.spring.mapper.MapperFactoryBean]: Bean property 'mapperInterface' is not writable or has an invalid setter method. Does the parameter type of the setter match the return type of the getter?
        at org.springframework.beans.BeanWrapperImpl.createNotWritablePropertyException(BeanWrapperImpl.java:243) ~[na:na]
        at org.springframework.beans.AbstractNestablePropertyAccessor.processLocalProperty(AbstractNestablePropertyAccessor.java:432) ~[na:na]
        at org.springframework.beans.AbstractNestablePropertyAccessor.setPropertyValue(AbstractNestablePropertyAccessor.java:278) ~[na:na]
        at org.springframework.beans.AbstractNestablePropertyAccessor.setPropertyValue(AbstractNestablePropertyAccessor.java:266) ~[na:na]
        at org.springframework.beans.AbstractPropertyAccessor.setPropertyValues(AbstractPropertyAccessor.java:104) ~[na:na]
        at org.springframework.beans.AbstractPropertyAccessor.setPropertyValues(AbstractPropertyAccessor.java:79) ~[na:na]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyPropertyValues(AbstractAutowireCapableBeanFactory.java:1740) ~[na:na]
        ... 36 common frames omitted
```

--- a/mybatis-spring-native-core/src/main/java/org/mybatis/spring/nativex/MapperFactoryBeanPostProcessor.java
+++ b/mybatis-spring-native-core/src/main/java/org/mybatis/spring/nativex/MapperFactoryBeanPostProcessor.java
@@ -1,0 +1,67 @@
+/*
+ *    Copyright 2022 the original author or authors.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package org.mybatis.spring.nativex;
+
+import org.mybatis.spring.mapper.MapperFactoryBean;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.BeanFactoryAware;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.beans.factory.support.RootBeanDefinition;
+import org.springframework.context.annotation.BeanDefinitionPostProcessor;
+import org.springframework.core.ResolvableType;
+import org.springframework.util.ClassUtils;
+
+class MapperFactoryBeanPostProcessor implements BeanDefinitionPostProcessor, BeanFactoryAware {
+
+  private static final String MAPPER_FACTORY_BEAN = "org.mybatis.spring.mapper.MapperFactoryBean";
+
+  private ConfigurableBeanFactory beanFactory;
+
+  @Override
+  public void setBeanFactory(BeanFactory beanFactory) throws BeansException {
+    this.beanFactory = (ConfigurableBeanFactory) beanFactory;
+  }
+
+  @Override
+  public void postProcessBeanDefinition(String beanName, RootBeanDefinition beanDefinition) {
+    if (ClassUtils.isPresent(MAPPER_FACTORY_BEAN, this.beanFactory.getBeanClassLoader())) {
+      resolveRepositoryFactoryBeanTypeIfNecessary(beanDefinition);
+    }
+  }
+
+  private void resolveRepositoryFactoryBeanTypeIfNecessary(RootBeanDefinition beanDefinition) {
+    if (!beanDefinition.hasBeanClass() || !MapperFactoryBean.class.isAssignableFrom(beanDefinition.getBeanClass())) {
+      return;
+    }
+    if (beanDefinition.getResolvableType().hasUnresolvableGenerics()) {
+      Class<?> mapperInterface = getMapperInterface(beanDefinition);
+      if (mapperInterface != null) {
+        beanDefinition.setTargetType(ResolvableType.forClassWithGenerics(
+                beanDefinition.getBeanClass(), mapperInterface));
+      }
+    }
+  }
+
+  private Class<?> getMapperInterface(RootBeanDefinition beanDefinition) {
+    try {
+      return (Class<?>) beanDefinition.getPropertyValues().get("mapperInterface");
+    } catch (Exception ex) {
+      return null;
+    }
+  }
+
+}

--- a/mybatis-spring-native-core/src/main/resources/META-INF/spring.factories
+++ b/mybatis-spring-native-core/src/main/resources/META-INF/spring.factories
@@ -1,6 +1,8 @@
 org.springframework.aot.context.bootstrap.generator.infrastructure.nativex.BeanFactoryNativeConfigurationProcessor=\
   org.mybatis.spring.nativex.MyBatisMapperNativeConfigurationProcessor,\
   org.mybatis.spring.nativex.MyBatisScannedResourcesNativeConfigurationProcessor
+org.springframework.context.annotation.BeanDefinitionPostProcessor=\
+  org.mybatis.spring.nativex.MapperFactoryBeanPostProcessor
 org.springframework.nativex.type.NativeConfiguration=\
   org.mybatis.spring.nativex.MyBatisCoreNativeConfiguration,\
   org.mybatis.spring.nativex.MyBatisSpringNativeConfiguration

